src/main/cpp/autoalignment/Autoalignment.cpp

```diff
@@ -1,7 +1,165 @@
+        (blueAlliancePastChargingStationThresholdYPositive +
+         blueAlliancePastChargingStationThresholdYNegative) /
+        2.0;
+
+    const units::meter_t targetSideOfChargingStationYPos =
+        currentPose.Y() > chargingStationHorizontalBisectionLineY
+            ? blueAlliancePastChargingStationThresholdYPositive
+            : blueAlliancePastChargingStationThresholdYNegative;
+
+    if (currentPose.X() > blueAlliancePastChargingStationThresholdX) {
+      const units::meter_t targetXPos =
+          blueAlliancePastChargingStationThresholdX;
+
+      pathPoints.push_back(pathplanner::PathPoint(
+          frc::Translation2d(targetXPos, targetSideOfChargingStationYPos),
+          frc::Rotation2d(0.0_deg)));
+    }
+
+    if (currentPose.X() > blueAllianceBeforeChargingStationThresholdX) {
+      const units::meter_t targetXPos =
+          blueAllianceBeforeChargingStationThresholdX;
+      pathPoints.push_back(pathplanner::PathPoint(
+          frc::Translation2d(targetXPos, targetSideOfChargingStationYPos),
+          frc::Rotation2d(0.0_deg)));
+    }
+
+    pathPoints.push_back(pathplanner::PathPoint(
+        frc::Translation2d(targetPose.X(), targetPose.Y()),
+        frc::Rotation2d(targetPose.Rotation())));
+  } else if (team == frc::DriverStation::Alliance::kRed) {
+    auto currentPose = driveSubsystem.getPose();
+    const units::meter_t chargingStationHorizontalBisectionLineY =
+        (redAlliancePastChargingStationThresholdYPositive +
+         redAlliancePastChargingStationThresholdYNegative) /
+        2.0;
+
+    const units::meter_t targetSideOfChargingStationYPos =
+        currentPose.Y() > chargingStationHorizontalBisectionLineY
+            ? redAlliancePastChargingStationThresholdYPositive
+            : redAlliancePastChargingStationThresholdYNegative;
+
+    if (currentPose.X() < redAlliancePastChargingStationThresholdX) {
+      const units::meter_t targetXPos =
+          redAlliancePastChargingStationThresholdX;
+
+      pathPoints.push_back(pathplanner::PathPoint(
+          frc::Translation2d(targetXPos, targetSideOfChargingStationYPos),
+          frc::Rotation2d(0.0_deg)));
+    }
+
+    if (currentPose.X() < redAllianceBeforeChargingStationThresholdX) {
+      const units::meter_t targetXPos =
+          redAllianceBeforeChargingStationThresholdX;
+      pathPoints.push_back(pathplanner::PathPoint(
+          frc::Translation2d(targetXPos, targetSideOfChargingStationYPos),
+          frc::Rotation2d(0.0_deg)));
+    }
+
+    pathPoints.push_back(pathplanner::PathPoint(
+        frc::Translation2d(targetPose.X(), targetPose.Y()),
+        frc::Rotation2d(targetPose.Rotation())));
+  } else {
+    std::cerr << "invalid alliance!" << std::endl;
+  }
+
+  pathplanner::PathPlannerTrajectory trajectory =
+      pathplanner::PathPlanner::generatePath(
+          pathplanner::PathConstraints(4_mps, 3_mps_sq), pathPoints);
+
+  pathplanner::RamseteAutoBuilder builder(
```

Comment:
You should pass the auto builder from the robot container as a a parameter to the function since constructing It here will cause it to go out of scope. 

Also, the automatic path transform for alliance won't work for us since the field origin changes depending on which alliance you are. It kinda made sense when field were rotationally symmetric, but even then that decision seems iffy. I put in a pull request to change this in PathPlanner, but this coordinate system is the official WPILib way so it should be rejected (even though Shuffleboard doesn't seem follow that standard, Aaargghh!).

We'll have to add a functions to transform paths the way we want since our origin must be alliance independent for vision to work.